12장 : 1,2,3,5,10

분산 분석을 하는 모든 데이터는 다음과 같은 3개의 조항을 만족한다고 가정을 한다.

1. 비교대상인 모집단의 자료는 각각 정규분포를 따른다.

2. 각 비교 대상 모집단의 자료는 동일한 분산을 가진다.

3. 각 표본은 독립적으로 추출되었다.

In [31]:
import pandas as pd
import numpy as np
import warnings
import seaborn as sns; sns.set()
import matplotlib.pyplot as plt
from statsmodels.formula.api import ols
from statsmodels.stats.anova import anova_lm
warnings.filterwarnings(action='ignore')

#### 12.1
H0: 기계간 평균 인장강도가 모두 같다.

H1:기계간 평균 인장강도가 적어도 하나는 다르다.

In [32]:
g1 = [17.5,16.9,15.8,18.6] 
g2 = [16.4,19.2,17.7,15.4]
g3 = [20.3,15.7,17.8,18.9]
g4 = [14.6,16.7,20.8,18.9]
g5 = [17.5,19.2,16.5,20.5]
g6 = [18.3,16.2,17.5,20.1]

data=pd.DataFrame({'g1':g1,'g2':g2,'g3':g3,'g4':g4,'g5':g5,'g6':g6})
data

g1    g2    g3    g4    g5    g6
0  17.5  16.4  20.3  14.6  17.5  18.3
1  16.9  19.2  15.7  16.7  19.2  16.2
2  15.8  17.7  17.8  20.8  16.5  17.5
3  18.6  15.4  18.9  18.9  20.5  20.1

In [33]:
y = ['g1','g1','g1','g1','g2','g2','g2','g2',
    'g3','g3','g3','g4','g4','g4','g4','g4',
     'g5','g5','g5','g5','g6','g6','g6','g6']
x= [17.5,16.9,15.8,18.6,
16.4,19.2,17.7,15.4,
20.3,15.7,17.8,18.9,
14.6,16.7,20.8,18.9,
17.5,19.2,16.5,20.5,
18.3,16.2,17.5,20.1]
data=pd.DataFrame({"value":x,"group":y})
data.head(6)

value group
0   17.5    g1
1   16.9    g1
2   15.8    g1
3   18.6    g1
4   16.4    g2
5   19.2    g2

In [34]:
model = ols('value ~ C(group)', data).fit()
anova_lm(model)

df     sum_sq   mean_sq         F   PR(>F)
C(group)   5.0   4.981167  0.996233  0.284651  0.91539
Residual  18.0  62.997167  3.499843       NaN      NaN

under H0~

PR(>F) : 0.91 > 유의 수준 (0.05)

즉,

p-value > a : H1 를 기각, H0 의 주장을 뒷받침하는 증거라 말 할수있다.

결론:기계간 평균 인장강도가 모두 같다라고 할 수 있다.

#### 12 2

H0: 진통제간 평균 진통시간이 같다

H1:진통제간 평균 지농시간이 적어도 하나는 다르다.

In [35]:
y=[]
for x in range(5):
    y.append("A")
for x in range(5):
    y.append("B")
for x in range(5):
    y.append("C")
for x in range(5):
    y.append("D")
for x in range(5):
    y.append("E")
x = [5.2,4.7,8.1,6.2,3.0,
     9.1,7.1,8.2,6.0,9.1,
     3.2,5.8,2.2,3.1,7.2,
     2.4,3.4,4.1,1.0,4.0,
     7.1,6.6,9.3,4.2,7.6
    ]

data=pd.DataFrame({"value":x,"group":y})
data.head(6)

model = ols('value ~ C(group)', data).fit()
anova_lm(model)

df   sum_sq  mean_sq         F    PR(>F)
C(group)   4.0  78.4216  19.6054  6.586508  0.001497
Residual  20.0  59.5320   2.9766       NaN       NaN

under H0~

PR(>F) : 0.001497 < 유의 수준 (0.05)

즉,
p-value < a : H0 를 기각, H1 의 주장을 뒷받침하는 증거라 말 할 수 있다.

결론:진통제간 평균 진통 시간이 적어도 하나는 다르다고 할 수 있다.

#### 12 3

H0: 선반 높이에따라 일일평균 매출액이 같다.
m1=m2=m3

H1: not h0


In [36]:
y = []
for x in range(8):
    y.append("L")
for x in range(8):
    y.append("M")
for x in range(8):
    y.append("H")

x = [77,82,86,78,81,86,77,81,
    88,94,93,90,91,94,90,87,
    85,85,87,81,80,79,87,93
    ]

data=pd.DataFrame({"value":x,"group":y})
data.head(10)

value group
0     77     L
1     82     L
2     86     L
3     78     L
4     81     L
5     86     L
6     77     L
7     81     L
8     88     M
9     94     M

In [37]:
model = ols('value ~ C(group)', data).fit()
anova_lm(model)

df  sum_sq  mean_sq          F   PR(>F)
C(group)   2.0  399.25  199.625  14.518182  0.00011
Residual  21.0  288.75   13.750        NaN      NaN

under H0~

PR(>F) : 0.0001 < 유의 수준 (0.01)

즉,
p-value < a : H0 를 기각, H1 의 주장을 뒷받침하는 증거라 말 할 수 있다.

결론:선반 높이에따른 일일평균 매출액이 적어도 하나는 다르다고 할 수 있다.

#### 12 5

H0: 네 가지 농도에 대하여 평균활성이 같다
m1=m2=m3=m4

H1: not h0

In [40]:
y = []
for x in range(4):
    y.append("0")
for x in range(4):
    y.append("80")
for x in range(4):
    y.append("160")
for x in range(4):
    y.append("160")

x=[11.01,12.09,10.55,11.26,11.38,10.67,12.33,10.08,11.02,10.67,11.50,
  10.31,6.04,8.65,7.76,10.13]
data=pd.DataFrame({"value":x,"group":y})
model = ols('value ~ C(group)', data).fit()
anova_lm(model)

df     sum_sq   mean_sq         F    PR(>F)
C(group)   2.0  11.064319  5.532159  2.521656  0.118734
Residual  13.0  28.520175  2.193860       NaN       NaN

under H0~
PR(>F) : 0.11 > 유의 수준 (0.01)

즉,
p-value > a : H1 를 기각, H0 의 주장을 뒷받침하는 증거라 말 할 수 있다.

결론:네 가지 농도에 대하여 평균활성이 같다

#### 12 10 

12.4의 자료 이용 0.01 Cochran 등분산검정


H0: 3종류의 부동화제의 따른 평균 압도 시간이 같다
m1=m2=m3

H1: 3종류의 부동화제의 따른 평균 압도 시간이 적어도 하나는 다르다.

In [44]:
y = []
for x in range(10):
    y.append("A")
for x in range(10):
    y.append("B")
for x in range(10):
    y.append("C")
    
x=[11,5,14,7,10,7,23,4,11,11,
    10,7,16,7,7,5,10,10,6,12,
    4,4,6,3,5,6,8,3,7,3
  ]
data=pd.DataFrame({"value":x,"group":y})
model = ols('value ~ C(group)', data).fit()
anova_lm(model)

df      sum_sq    mean_sq         F    PR(>F)
C(group)   2.0  158.866667  79.433333  5.457252  0.010222
Residual  27.0  393.000000  14.555556       NaN       NaN

under H0~
PR(>F) : 0.010222 > 유의 수준 (0.01)

즉,
p-value > a : H1 를 기각, H0 의 주장을 뒷받침하는 증거라 말 할 수 있다.

결론:3종류의 부동화제의 따른 평균 압도 시간의 차이가 없다.